client/src/components/Workflow/Editor/NodeOutput.vue

```diff
@@ -122,7 +122,7 @@ function onToggleActive() {
 
 function onToggleVisible() {
     const actionKey = `HideDatasetAction${props.output.name}`;
-    const step = stepStore.getStep(stepId.value);
+    const step = { ...stepStore.getStep(stepId.value) };
     assertDefined(step);
```

Comment:
A more elegant solution than my previous suggestion.
Requires  #15834

```suggestion
    const step =  { ...ensureDefined(stepStore.getStep(stepId.value)) };
```